core/provider/provider.go

```diff
@@ -33,6 +27,13 @@ type Provider struct {
 	UpdatedAt time.Time              `bson:"updated_at"`
 }
 
+func (p *Provider) Validate() error {
+	p.URN = GenerateURN(*p)
+
+	// TODO: add basic sanitization and validations here.
+	return nil
+}
+
 func GenerateURN(pro Provider) string {
```

Comment:
Since this is now being used via `Validate` shall we make this private?

Reply:
yea. i tried. it was being used in some places in server package. so wanted to raise a separate PR that completes the validate method and changes this to private.